server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +342,126 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
+                        );
+                        return null;
+                      }
+                    }
+                )
+            );
+          }
+          final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+          try {
+            futureList.get()
+                      .stream()
+                      .filter(Objects::nonNull)
+                      .forEach(builder::put);
+          }
+          catch (Exception ex) {
+            futureList.cancel(true);
+            throw ex;
+          }
+          stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+        }
+        catch (Exception e) {
+          LOG.error(e, "Failed to finish lookup load process.");
+        }
+        executorService.shutdownNow();
+      } else {
+        LOG.info("No lookups to be loaded at this point");
+        stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+      }
+    } else {
+      LOG.info("Lookup loading on startup is disabled");
+      stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+    }
+  }
 
-        if (container.getLookupExtractorFactory().start()) {
-          builder.put(lookupBean.getName(), container);
+  private List<LookupBean> getLookupListFromCoordinator(String tier)
+  {
+    try {
+      final FullResponseHolder response = fetchLookupsForTier(tier);
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+        LOG.error(
+            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
+            response.getStatus(),
+            response.getContent()
+        );
+        return null;
+      } else {
+        // Older version of getSpecificTier returns a list of lookup names.
+        // Lookup loading is performed via snapshot if older version is present.
+        // This check is only for backward compatibility and should be removed in a future release
+        if (response.getContent().startsWith("[")) {
+          return null;
         } else {
-          throw new ISE("Failed to start lookup [%s]:[%s]", lookupBean.getName(), container);
+          Map<String, LookupExtractorFactoryContainer> lookupMap = jsonMapper.readValue(
+              response.getContent(),
+              LOOKUPS_ALL_REFERENCE
+          );
+          if (!lookupMap.isEmpty()) {
+            for (Map.Entry<String, LookupExtractorFactoryContainer> e : lookupMap.entrySet()) {
+              lookupBeanList.add(new LookupBean(e.getKey(), null, e.getValue()));
+            }
+          }
         }
       }
+      return lookupBeanList;
+    }
+    catch (ISE ise) {
+      LOG.error(ise, "Coordinator is unavailable");
+      return null;
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error while parsing lookup code for tier [%s] from response", tier);
```

Comment:
I would change the message to something generic like
```
Error while trying to get lookup list from coordinator for tier[%s]
```
exception would already contain specific details.

and remove the other catch block covering ISE specially.

Reply:
I have the ISE specific catch block so that we understand it is the case where coordinator is unavailable.